graylog2-server/src/main/java/org/graylog2/periodical/IndexRangesCleanupPeriodical.java

```diff
@@ -0,0 +1,131 @@
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.periodical;
+
+import com.github.joschi.jadconfig.util.Duration;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.Ints;
+import com.google.inject.name.Named;
+import org.graylog2.indexer.Deflector;
+import org.graylog2.indexer.cluster.Cluster;
+import org.graylog2.indexer.esplugin.IndicesDeletedEvent;
+import org.graylog2.indexer.ranges.IndexRange;
+import org.graylog2.indexer.ranges.IndexRangeService;
+import org.graylog2.plugin.periodical.Periodical;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link Periodical} to clean up stale index ranges (e. g. because the index has been deleted externally)
+ *
+ * @since 1.3.0
+ */
+public class IndexRangesCleanupPeriodical extends Periodical {
+    private static final Logger LOG = LoggerFactory.getLogger(IndexRangesCleanupPeriodical.class);
+
+    private final Cluster cluster;
+    private final Deflector deflector;
+    private final IndexRangeService indexRangeService;
+    private final EventBus eventBus;
+    private final int periodSeconds;
+
+    @Inject
+    public IndexRangesCleanupPeriodical(final Cluster cluster,
+                                        final Deflector deflector,
+                                        final IndexRangeService indexRangeService,
+                                        final EventBus eventBus,
+                                        @Named("index_ranges_cleanup_interval") final Duration indexRangesCleanupInterval) {
+        this.cluster = requireNonNull(cluster);
+        this.deflector = requireNonNull(deflector);
+        this.indexRangeService = requireNonNull(indexRangeService);
+        this.eventBus = requireNonNull(eventBus);
+        this.periodSeconds = Ints.saturatedCast(indexRangesCleanupInterval.toSeconds());
+
+    }
+
+    @Override
+    public void doRun() {
+        if (!cluster.isConnected() || !cluster.isHealthy()) {
+            LOG.info("Skipping index range cleanup because the Elasticsearch cluster is unreachable or unhealthy");
+            return;
+        }
+
+        final Set<String> indexNames = ImmutableSet.copyOf(deflector.getAllDeflectorIndexNames());
+        final SortedSet<IndexRange> indexRanges = indexRangeService.findAll();
+
+        final List<String> removedIndices = new ArrayList<>();
+        for (IndexRange indexRange : indexRanges) {
+            if (!indexNames.contains(indexRange.indexName())) {
+                removedIndices.add(indexRange.indexName());
+            }
+        }
+
+        if (!removedIndices.isEmpty()) {
+            LOG.info("Removing index range information for non-existing indices: {}", removedIndices);
```

Comment:
Should we change the wording a bit here? This also gets logged for closed indices.

```
2015-11-11 17:17:43,701 INFO : org.graylog2.periodical.IndexRangesCleanupPeriodical - Removing index range information for non-existing indices: [graylog2_2]
```